src/ray/object_manager/plasma/store.cc

```diff
@@ -355,9 +356,17 @@ void PlasmaStore::DisconnectClient(const std::shared_ptr<Client> &client) {
   create_request_queue_.RemoveDisconnectedClientRequests(client);
 }
 
-Status PlasmaStore::ProcessMessage(const std::shared_ptr<Client> &client,
-                                   fb::MessageType type,
-                                   const std::vector<uint8_t> &message) {
+void PlasmaStore::HandleClientConnectionError(const std::shared_ptr<Client> &client,
+                                              const boost::system::error_code &error) {
+  absl::MutexLock lock(&mutex_);
+  RAY_LOG(WARNING) << "Disconnecting client due to connection error with code "
```

Comment:
I would move logging out of critical section if possible.

Reply:
Logging could invoke IO on critical path, leading to unintended latency (I met if before).

Reply:
this is not in a critical section

Reply:
What do you mean? Lock is acquired one line above?

Reply:
I meant this is a rare failure-handling case, not in a critical section of the code. micro-optimizing performance here is not important

Reply:
Yeah you could argue that, but it's generally a good **coding practice** to put IO out of critical section

Reply:
i don't think it's a blocking issue, just raise it here, up to you

Reply:
agreed